### PR TITLE
Enable single species selection, fix bugs

### PR DESCRIPTION
**Pull request by YazidZidane**
- Enabled single species selection, hide piecharts in Mix Ratios and Review Mix if only one species is selected.
- Updated diversity bar to be shown based on numbers of species selected.
- Fixed a bug that mixRatioOption is not updated when a crop get removed.
- Fixed a bug that the app will crash if history was made only for first 2 steps.
- Fixed a bug that percentOdRate value is not updated if make any changes in species selection.

### CODE REVIEW

File: src/components/StepsList/index.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ const StepsList = ({
       return 'Please enter the necessary info below.';
     }
     if (activeStep === 1 && !availableSteps[1]) {
-      return 'Please select at least 2 plants.';
+      return 'Please select at least 1 plants.';
```
Comment (mikahpinegar) [inline src/components/StepsList/index.js:77]: should be `Please select at least 1 plant`

**Review by mikahpinegar — CHANGES_REQUESTED** [review]

File: src/components/StepsList/index.js (hunk 1 above)
Comment (YazidZidane) [reply src/components/StepsList/index.js]: I have updated this, I also noticed that the caption text on the page should change, what about `Select your species.`?
image

Comment (mikahpinegar) [reply src/components/StepsList/index.js]: Select one or more species for your mix.

**Comment by Adamws33** [comment]
when we are removing the pie charts from the mix ratios page when there is only one seed can we force the accordion to be open.  the page looks rather empty and doesn't seem intuitive on what is there.

**Comment by Adamws33** [comment]
> when we are removing the pie charts from the mix ratios page when there is only one seed can we force the accordion to be open. the page looks rather empty and doesn't seem intuitive on what is there.

same with the review mix page.

**Review by Adamws33 — APPROVED** [review]

**Review by mikahpinegar — APPROVED** [review]

**Pull request merged**